src/NHibernate.Test/DialectTest/MsSql2008DialectFixture.cs

```diff
@@ -148,16 +148,7 @@ public void ScaleTypes()
 			Assert.That(dialect.GetTypeName(SqlTypeFactory.GetTime(max + 1)), Is.EqualTo("time").IgnoreCase, "Over max time");
 		}
 
-		private static readonly FieldInfo _mappingField =
-			typeof(Configuration).GetField("mapping", BindingFlags.Instance | BindingFlags.NonPublic);
-
-		private static IMapping GetMapping(Configuration cfg)
-		{
-			Assert.That(_mappingField, Is.Not.Null, "Unable to find field mapping");
-			var mapping = _mappingField.GetValue(cfg) as IMapping;
-			Assert.That(mapping, Is.Not.Null, "Unable to find mapping object");
-			return mapping;
-		}
+		private static IMapping GetMapping(Configuration cfg) => (IMapping) cfg.BuildSessionFactory();
```

Comment:
The true equivalent for the original code in the new way would be `GetMapping(Configuration cfg) => cfg.BuildMapping()`, but `Dialect` would throw an `InvalidOperationException`. However, in this case it does not matter and we can use fully resolved session factory instead.

Reply:
Now new public "BuildMapping" can be used.

Reply:
I was thinking about it and don't like it. `cfg.BuildMapping` just should not exist as it provides an incomplete version of mapping, that can create a lot of problems.

Reply:
I don't understand your concern. In my understanding `IMapping` was specifically designed to be safely used in "uncompiled" state. See interface description:
https://github.com/nhibernate/nhibernate-core/blob/a0bc61f3a4de1fed25c4c2cbd6011227b5aa0468/src/NHibernate/Engine/IMapping.cs#L6-L7

So as long you use interface methods and don't do some funny castings you should be safe.

You agreed to keep method public - so it would be good to have it in tests too.

Reply:
I don't think the `IMapping` was designed. It was probably appeared incidentally from some other bad design decision. Problem with the interface that it is not clear when it is safe to use compiled or raw mappings. For example, `SingleTableEntityPersister` accepts both `IMapping` and `ISessionFactoryImplementor`. Can it use only `ISessionFactoryImplementor`? 

https://github.com/nhibernate/nhibernate-core/blob/3e353c1db3c8f106aa43f5d2f5b3b0c232785676/src/NHibernate/Persister/Entity/SingleTableEntityPersister.cs#L79-L81

This is really not clear.

Also, Hibernate has removed this interface and redesigned the mapping compilation.